config_lib/writers/writer_yaml.py

```diff
@@ -0,0 +1,46 @@
+import re
+
+
+def is_iso_datetime(value: str) -> bool:
+    iso_format = r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z$"
+    return isinstance(value, str) and re.match(iso_format, value)
+
+
+def format_yaml_value(value):
+    if isinstance(value, bool):
+        return "true" if value else "false"
+    elif value is None:
+        return "null"
+    elif isinstance(value, str):
+        return value if is_iso_datetime(value) else f'"{value}"'
+    else:
+        return str(value)
```

Comment:
Here as well, it's better to use 'if' statements instead:
```suggestion
def format_yaml_value(value):
    if isinstance(value, bool):
        return "true" if value else "false"

    if value is None:
        return "null"

    if isinstance(value, str):
        return value if is_iso_datetime(value) else f'"{value}"'

    return str(value)
```